SportsPlanner.Droid/MainActivity.cs

```diff
@@ -1,44 +1,63 @@
-using System;
-using System.Net.Http;
-using SportsPlanner.Entities;
-using SportsPlanner.Gateways;
-
 using Android.App;
 using Android.Widget;
 using Android.OS;
+using SportsPlanner.ViewModels;
+using System.Collections.Generic;
+using GalaSoft.MvvmLight.Helpers;
+using GalaSoft.MvvmLight.Views;
 
 namespace SportsPlanner.Droid
 {
-	[Activity(Label = "SportsPlanner.Droid", MainLauncher = true, Icon = "@drawable/icon")]
-	public class MainActivity : Activity
+    [Activity(Label = "SportsPlanner.Droid", MainLauncher = true, Icon = "@drawable/icon")]
+	public class MainActivity : ActivityBase
 	{
-		private IEventsGateway _eventsGateway;
+        // Keep track of bindings to avoid premature garbage collection
+        private readonly List<Binding> bindings = new List<Binding>();
+
+        // Gets a reference to the MainViewModel from the ViewModelLocator.
+        private MainViewModel Vm
+        {
+            get
+            {
+                return App.Locator.MainViewModel;
+            }
+        }
+
+        private TextView _pageTitle;
+
+        public TextView PageTitle
+        {
+            get
+            {
+                return _pageTitle
+                    ?? (_pageTitle = FindViewById<TextView>(Resource.Id.pageTitle));
+            }
+        }
+
+        private Button _getEventsBtn;
+
+        public Button GetEventsBtn
+        {
+            get
+            {
+                return _getEventsBtn
+                    ?? (_getEventsBtn = FindViewById<Button>(Resource.Id.btn_getEvents));
+            }
+        }
 
 		protected override void OnCreate(Bundle bundle)
 		{
 			base.OnCreate(bundle);
-			_eventsGateway = new EventsApiGateway();
 
 			// Set our view from the "main" layout resource
 			SetContentView(Resource.Layout.Main);
 
-			// Get our button from the layout resource,
-			// and attach an event to it
-			Button button = FindViewById<Button>(Resource.Id.myButton);
+            GetEventsBtn.SetCommand("Click", Vm.GetEventsCommand);
```

Comment:
Maybe we can avoid the hardcoded strings by `nameOf`
https://msdn.microsoft.com/en-us/library/dn986596.aspx